src/Parser/Mage/Arcane/Modules/ManaChart/TabComponent/ManaLevelGraph.js

```diff
@@ -0,0 +1,227 @@
+
+        if (!deadBosses.includes(series.guid)) {
+          const health = item[1];
+          newSeries.data[secIntoFight] = health;
+
+          if (health === 0) {
+            deadBosses.push(series.guid);
+          }
+        }
+      });
+      bosses.push(newSeries);
+    });
+    const deathsBySecond = {};
+    if (this.state.bossHealth.deaths) {
+      this.state.bossHealth.deaths.forEach((death) => {
+        const secIntoFight = Math.floor((death.timestamp - start) / 1000);
+        if (death.targetID === this.props.actorId) {
+          deathsBySecond[secIntoFight] = true;
+        }
+      });
+    }
+
+    const fightDurationSec = Math.ceil((end - start) / 1000);
+    const labels = [];
+    for (let i = 0; i <= fightDurationSec; i += 1) {
+      labels.push(i);
+
+      manaBySecond[i] = manaBySecond[i] !== undefined ? manaBySecond[i] : null;
+      bosses.forEach((series) => {
+        series.data[i] = series.data[i] !== undefined ? series.data[i] : null;
+      });
+      deathsBySecond[i] = deathsBySecond[i] !== undefined ? deathsBySecond[i] : undefined;
+    }
+
+    const data = {
+      labels,
+      datasets: [
+        ...bosses.map((series, index) => ({
+          label: `${series.name} Health`,
+          ...ManaStyles[`Boss-${index}`],
+          ...ManaStyles[`Boss-${series.guid}`],
+          data: Object.keys(series.data).map(key => series.data[key]),
+        })),
+        {
+          label: `Deaths`,
+          ...ManaStyles.Deaths,
+          data: Object.keys(deathsBySecond).map(key => deathsBySecond[key]),
+        },
+        {
+          label: `Mana`,
+          ...ManaStyles.Mana,
+          data: Object.keys(manaBySecond).map(key => manaBySecond[key]),
+        },
+      ],
+    };
+
+    const gridLines = ManaStyles.gridLines;
+
+    const options = {
+      responsive: true,
+      scales: {
+        yAxes: [{
+          gridLines: gridLines,
+          ticks: {
+            callback: (value, index, values) => {
+              return `${value}%`;
+            },
+            min: 0,
+            max: 100,
+            stepSize: 25,
+            fontSize: 14,
+          },
+        }],
+        xAxes: [{
+          gridLines: gridLines,
+          ticks: {
+            callback: (value, index, values) => {
+              if(value%30 === 0) {
+                return formatDuration(value);
+              }
+              return null;
+            },
+            fontSize: 14,
+          },
+        }],
+      },
+      animation: {
+        duration: 0,
+      },
+      hover: {
+        animationDuration: 0,
+      },
+      responsiveAnimationDuration: 0,
+      tooltips: {
+        enabled: false,
+      },
+      legend: ManaStyles.legend,
+    };
+
+    Chart.plugins.register({
+      id: 'specialEventIndiactor',
+      afterDatasetsDraw : (chart) => {
+        const ctx = chart.ctx;
+
+        chart.data.datasets.forEach(function(dataset, i) {
```

Comment:
is there a reason you went with this approach for drawing deaths rather than the one used in the [StaggerPool](https://github.com/WoWAnalyzer/WoWAnalyzer/blob/master/src/Parser/Monk/Brewmaster/Modules/Features/StaggerPoolGraph.js#L148-L155)?

Reply:
oh, gotcha. i can go ahead and approve if you like or you can fix, up to you

Reply:
ill look at the staggerpool one to see what that looks like. 

Ideally id like to figure out how to draw stuff on the graph cause id like to be able to show where lust was cast and when arcane power was cast ... so if the staggerpool one is better then maybe later i can use that to add stuff.

Reply:
So from what I can tell (with the very limited knowledge I have).

I would need to redo the entire thing because StaggerPool uses the on_toPlayer_death, on_toPlayer_damage, etc stuff which i believe wont work in the existing ManaLevelChart cause it extends React.PureComponent instead of Analyzer? Not sure if thats true or not, but the on_toPlayer_death stuff doesnt do anything if i put it in the ManaLevelGraph stuff.

So I think in order to do it that way I would need to completely start from scratch and build the chart myself (which would involve trying to figure out how to do the WCL Query that it is doing to get the Boss Health instead of Max HP like you are using and getting mana instead of hp. At the moment I just dont know enough to be able to do that without a ton of time to spend hours and hours ripping stuff out and redoing it to try and stumble my way through it. 

So I think for now I should stick to the one I have that works, and then once arcane is fully supported and bfa ready, then maybe i can tackle trying to make a chart from scratch to be able to accomodate things like showing the arcane power casts and lust and such.

Reply:
👍